compiler/qsc/src/interpret.rs

```diff
@@ -639,6 +672,45 @@ impl Interpreter {
         })
     }
 
+    /// Performs QIR codegen using the given callable with the given arguments on a new instance of the environment
+    /// and simulator but using the current compilation.
+    pub fn qirgen_from_callable(
+        &mut self,
+        callable: &Value,
+        args: Value,
+    ) -> std::result::Result<String, Vec<Error>> {
+        if self.capabilities == TargetCapabilityFlags::all() {
+            return Err(vec![Error::UnsupportedRuntimeCapabilities]);
+        }
+
+        let Value::Global(store_item_id, _) = callable else {
+            panic!("value is not a global callable");
```

Comment:
we return a `Result` here, so this function is fallible -- any reason to use `panic!()` over an error?

Reply:
Good question... I was thinking about how this is called by the Python layer specifically, but this could do a more normal failure. This one doesn't really need to be a panic.